crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -2434,6 +2438,27 @@ impl RpcServerHandle {
         let client = builder.build(url).await.expect("failed to create ws client");
         Some(client)
     }
+
+    /// Returns an http provider from the rpc server handle
+    pub fn new_http_provider(&self) -> Result<RootProvider<Http<reqwest::Client>>, eyre::Error> {
+        let rpc_url = self.http_url().unwrap().parse()?;
+        Ok(ProviderBuilder::new().on_http(rpc_url))
+    }
```

Comment:
I'd like to make this a bit more useful, because currently all of these just return a regular provider, no recommended fillers, etc. and also tied to the Ethereum network.

we can introduce a few more helper functions that accept the Network as generic for example

and I think by default the recommended fillers should be installed

Reply:
Can you please leave some pointers regarding the additional helper functions mentioned.

Reply:
@mattsse  Are the helper functions needed to cater to the various fillers?
For eg: Different helper functions for gas_filler, nonce_filler etc. and by default using the recommended filler.
Or is there some other functionality you are asking for?